### PR TITLE
Python testing

### PR DESCRIPTION
**Pull request by joaohts**
Added prompts for python. Changed how block pythons are interpreted: now one block of python runs until a return is found. Overall changes to make run_whole.py work with python.

### CODE REVIEW

File: default_prompts.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
     CHECK_CHEAT = NO_CHECK_CHEAT
 
 problem_multistep_python = (
-    f"""### Spec: In {LANG}, write a function  that returns the sum of the unique prime factors of 
+    f"""### Spec: In {LANG}, write a function sum_unique_prime_factors(n) that returns the sum of the unique prime factors of 
 a given natural number. Example: if the input number was 28, the function would return 9 (2 + 7),
```
Comment (namin) [inline default_prompts.py:33]: do you want to mention n as the given natural number>

File: default_prompts.py [hunk 2]
```diff
@@ -40,9 +40,142 @@
...
     ['Python'],
     None,
-    None
+    {
+        "def sum_unique_prime_factors(n):": "test(sum_unique_prime_factors(28) == 9)\ntest(sum_unique_prime_factors(45) == 8)\ntest(sum_unique_prime_factors(60) == 10)"
+    }
+)
+
+problem_python_2_oneshot = (
+    f"""### Spec: In {LANG}, write a 'magic(s)' function that: given a string, it finds all the characters that repeat an odd
```
Comment (namin) [inline default_prompts.py:49]: Do you want to use quotes like this `magic`? Why say s explicitly?

File: default_prompts.py [hunk 3]
```diff
@@ -40,9 +40,142 @@
...
+    NO_CHECK_PROOF, NO_CHECK_CHEAT,
+    ['Python'],
+    None,
+    {
+        "def magic(s):": "test(magic('vvbbbsszzzzcc  kklllccffzzrr!!') == 'hello world!')"
+    }
+)
+
+problem_python_2_steps = (
+    f"""### Spec: In {LANG}, write a function 'remove_odd(s)', that given a string, it removes the characters that
```
Comment (namin) [inline default_prompts.py:68]: ditto for code quotes

File: default_prompts.py [hunk 4]
```diff
@@ -40,9 +40,142 @@
...
+    {
+        "def remove_odd(s):" : "test(remove_odd('kkkppoottaiiiattoojjj') == 'ppoottaattoo')",
+        "def remove_pairs(s):" : "test(remove_pairs('ppoottaattoo') == 'potato')\ntest(remove_pairs('hheelllloo') == 'hello')",
+        "def rotate_letters(s, n):" : "test(rotate_letters('abc', 1) == 'bcd')",
+        "def magic(s):" : "test(magic('vvbbbsszzzzcc  kklllccffzzrr!!') == 'hello world!')"
+    }
+)
+
+problem_python_3_oneshot = (
+    f"""### Spec: In {LANG}, Write a function string_to_number(s) that given a tring s, converts each of its alpha numeric characters into numbers. The convesion
```
Comment (namin) [inline default_prompts.py:92]: typo tring

File: default_prompts.py [hunk 5]
```diff
@@ -40,9 +40,142 @@
...
+    NO_CHECK_PROOF, NO_CHECK_CHEAT,
+    ['Python'],
+    None,
+    {
+        "def best_string(t):": "test(best_string('Hello world! A quick example with numbers 123 and special characters $%#.') == 'quick')"
+    }
+)
+
+problem_python_3_steps = (
+    f"""### Spec: In {LANG}, Write a function string_to_number(s) that given a tring s, converts each of its alpha numeric characters into numbers. The convesion
```
Comment (namin) [inline default_prompts.py:112]: ditto typo tring

File: python.py [hunk 6]
```diff
@@ -31,14 +31,14 @@ def find_first_index(string, char1, char2):
         index2 = float('inf')  # If char2 is not found, set to infinity
 
     if index1 == float('inf') and index2 == float('inf'):
-        return -1  # If neither character is found
+        return -1  # If neither character is founds
```
Comment (namin) [inline python.py:34]: founds should be found

**Review by namin — CHANGES_REQUESTED** [review]

File: default_prompts.py (hunk 1 above)
Comment (joaohts) [reply default_prompts.py]: good idea!

File: default_prompts.py (hunk 2 above)
Comment (joaohts) [reply default_prompts.py]: I removed the quotes from the magic. I think it is worthwhile to say magic(s) explicitly to force the LLM to generate the right name for the function. The unit tests depend on the right formatting of the function name.

**Review by namin — APPROVED** [review]

**Pull request merged**